src/tools/genpolicy/rules.rego

```diff
@@ -883,72 +882,114 @@ mount_source_allows(p_mount, i_mount, bundle_id, sandbox_id) {
 
     print("mount_source_allows 2: true")
 }
+mount_source_allows(p_mount, i_mount, bundle_id, sandbox_id) {
+    print("mount_source_allows 3: i_mount.source=", i_mount.source)
+
+    i_source_parts = split(i_mount.source, "/")
+    b64_device_id = i_source_parts[count(i_source_parts) - 1]
+
+    base64.is_valid(b64_device_id)
+
+    source1 := p_mount.source
+    print("mount_source_allows 3: source1 =", source1)
+
+    source2 := replace(source1, "$(spath)", policy_data.common.spath)
+    print("mount_source_allows 3: source2 =", source2)
+
+    source3 := replace(source2, "$(b64_device_id)", b64_device_id)
+    print("mount_source_allows 3: source3 =", source3)
+
+    source3 == i_mount.source
+
+    print("mount_source_allows 3: true")
+}
 
 ######################################################################
 # Create container Storages
 
 allow_storages(p_storages, i_storages, bundle_id, sandbox_id) {
-    p_count := count(p_storages)
-    i_count := count(i_storages)
+    # TODO: Remove this filtering when we implement generic storage 
```

Comment:
I don't think adding such complexity is justified. We should just delete the previously-unused allow_storages() and replace it with something small, that works just for the storage type you are adding here.

I will try to help implement a better allow_storages() if someone can explain to me its requirements, how to test it in a realistic CoCo environment, etc.